File: Scrabble-v2/web/src/app/services/sound.service.ts
import { Injectable } from '@angular/core';
import { SOUNDS_MAP } from '@app/constants/sounds-constant';

@Injectable({
    providedIn: 'root',
})
export class SoundService {
    turnOn: boolean = true;
    controllerSound(typeSound: string): void {
        const audio = new Audio();
        audio.src = './assets/sound_effects/' + SOUNDS_MAP.get(typeSound);
        audio.load();
        if (this.turnOn) {
            audio.volume = 0.1;
            audio.play();
        } else {
            audio.pause();
            audio.currentTime = 0;
        }
    }
}

File: Scrabble-v2/mobile/lib/services/sound_service.dart
import 'package:assets_audio_player/assets_audio_player.dart';
import 'package:flutter/cupertino.dart';

import '../constants/sound_effects.dart' as sound_effects;

class SoundService extends ChangeNotifier {
  bool turnOn = true;
  SoundService(this.turnOn);

  String findAudioSrc(String typeSound) {
    String prefixSrc = '/assets/audio/';
    return prefixSrc + sound_effects.SOUNDS_MAP[typeSound].toString();
  }

  Future<void> controllerSound(String typeSound) async {
    AssetsAudioPlayer audioplayer = AssetsAudioPlayer();
    if (turnOn) {
      audioplayer.open(Audio(findAudioSrc(typeSound)));
      audioplayer.setVolume(sound_effects.moderateVolumeLevel);
      await audioplayer.play();
    } else {
      await audioplayer.stop();
    }
  }
}
